python/graphframes/tutorials/stackexchange.py

```diff
@@ -0,0 +1,595 @@
+        .otherwise("Unknown"),
+    )
+    .withColumn("Type", F.lit("PostLinks"))
+)
+
+
+#
+# Load the PostHistory...
+#
+
+post_history_df = (
+    spark.read.format("xml")
+    .options(rowTag="row")
+    .options(rootTag="posthistory")
+    .load(f"{BASE_PATH}/PostHistory.xml")
+)
+print(f"Total PostHistory: {post_history_df.count():,}")
+
+# Remove the _ prefix from field names
+post_history_df = remove_prefix(post_history_df).withColumn("Type", F.lit("PostHistory"))
+
+
+#
+# Load the Comments...
+#
+
+comments_df = (
+    spark.read.format("xml")
+    .options(rowTag="row")
+    .options(rootTag="comments")
+    .load(f"{BASE_PATH}/Comments.xml")
+)
+print(f"Total Comments:    {comments_df.count():,}")
+
+# Remove the _ prefix from field names
+comments_df = remove_prefix(comments_df).withColumn("Type", F.lit("Comment"))
+
+
+#
+# Load the Users...
+#
+
+users_df = (
+    spark.read.format("xml")
+    .options(rowTag="row")
+    .options(rootTag="users")
+    .load(f"{BASE_PATH}/Users.xml")
+)
+print(f"Total Users:       {users_df.count():,}")
+
+# Remove the _ prefix from field names
+users_df = remove_prefix(users_df).withColumn("Type", F.lit("User"))
+
+
+#
+# Load the Votes...
+#
+
+votes_df = (
+    spark.read.format("xml")
+    .options(rowTag="row")
+    .options(rootTag="votes")
+    .load(f"{BASE_PATH}/Votes.xml")
+)
+print(f"Total Votes:       {votes_df.count():,}")
+
+# Remove the _ prefix from field names
+votes_df = remove_prefix(votes_df).withColumn("Type", F.lit("Vote"))
+
+# Add a VoteType column
+votes_df = votes_df.withColumn(
+    "VoteType",
+    F.when(F.col("VoteTypeId") == 2, "UpVote")
+    .when(F.col("VoteTypeId") == 3, "DownVote")
+    .when(F.col("VoteTypeId") == 4, "Favorite")
+    .when(F.col("VoteTypeId") == 5, "Close")
+    .when(F.col("VoteTypeId") == 6, "Reopen")
+    .when(F.col("VoteTypeId") == 7, "BountyStart")
+    .when(F.col("VoteTypeId") == 8, "BountyClose")
+    .when(F.col("VoteTypeId") == 9, "Deletion")
+    .when(F.col("VoteTypeId") == 10, "Undeletion")
+    .when(F.col("VoteTypeId") == 11, "Spam")
+    .when(F.col("VoteTypeId") == 12, "InformModerator")
+    .otherwise("Unknown"),
+)
+
+
+#
+# Load the Tags...
+#
+
+tags_df = (
+    spark.read.format("xml")
+    .options(rowTag="row")
+    .options(rootTag="tags")
+    .load(f"{BASE_PATH}/Tags.xml")
+)
+print(f"Total Tags:        {tags_df.count():,}")
+
+# Remove the _ prefix from field names
+tags_df = remove_prefix(tags_df).withColumn("Type", F.lit("Tag"))
+
+
+#
+# Load the Badges...
+#
+
+badges_df = (
+    spark.read.format("xml")
+    .options(rowTag="row")
+    .options(rootTag="badges")
+    .load(f"{BASE_PATH}/Badges.xml")
+)
+print(f"Total Badges:      {badges_df.count():,}\n")
+
+# Remove the _ prefix from field names
+badges_df = remove_prefix(badges_df).withColumn("Type", F.lit("Badge"))
+
+
+#
+# Form the nodes from the UNION of posts, users, votes and their combined schemas
+#
+
+all_cols: List[Tuple[str, T.StructField]] = list(
```

Comment:
Let's add `from __future__ import annotations` and use `list[tuple[str, T.StructField]]` instead?